src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
We have a purgekeys API for deleting principals.

We don't have to completely design this kvno feature; the point is that it's entirely reasonable to demand an explicit kvno update after (1) setting an explicit kvno on a principle, and (2) using chrand/chpass/setkey -keepold.

We have a purgekeys API for removing whole kvnos.  When it was developed, we noted at the time that Heimdal could remove kvnos from an existing key set, but Tom elected not to provide that functionality.  My understanding is that in Heimdal it's normal to create keys for a principal and then prune its enctypes of any values you don't want for that service, while in MIT krb5 you normally create the keys with the desired enctypes in the first place.  So we may just not have any use cases for that functionality.